examples/esphttpd/user_main.c

```diff
@@ -0,0 +1,208 @@
+	.tagName=LIBESPHTTPD_OTA_TAGNAME
+};
+
+
+/*
+This is the main url->function dispatching data struct.
+In short, it's a struct with various URLs plus their handlers. The handlers can
+be 'standard' CGI functions you wrote, or 'special' CGIs requiring an argument.
+They can also be auth-functions. An asterisk will match any url starting with
+everything before the asterisks; "*" matches everything. The list will be
+handled top-down, so make sure to put more specific rules above the more
+general ones. Authorization things (like authBasic) act as a 'barrier' and
+should be placed above the URLs they protect.
+*/
+HttpdBuiltInUrl builtInUrls[]={
+	{"*", cgiRedirectApClientToHostname, "esp8266.nonet"},
+	{"/", cgiRedirect, "/index.tpl"},
+	{"/led.tpl", cgiEspFsTemplate, tplLed},
+	{"/index.tpl", cgiEspFsTemplate, tplCounter},
+	{"/led.cgi", cgiLed, NULL},
+#ifndef ESP32
+	{"/flash/", cgiRedirect, "/flash/index.html"},
+	{"/flash/next", cgiGetFirmwareNext, &uploadParams},
+	{"/flash/upload", cgiUploadFirmware, &uploadParams},
+	{"/flash/reboot", cgiRebootFirmware, NULL},
+#endif
+	//Routines to make the /wifi URL and everything beneath it work.
+
+//Enable the line below to protect the WiFi configuration with an username/password combo.
+//	{"/wifi/*", authBasic, myPassFn},
+
+	{"/wifi", cgiRedirect, "/wifi/wifi.tpl"},
+	{"/wifi/", cgiRedirect, "/wifi/wifi.tpl"},
+	{"/wifi/wifiscan.cgi", cgiWiFiScan, NULL},
+	{"/wifi/wifi.tpl", cgiEspFsTemplate, tplWlan},
+	{"/wifi/connect.cgi", cgiWiFiConnect, NULL},
+	{"/wifi/connstatus.cgi", cgiWiFiConnStatus, NULL},
+	{"/wifi/setmode.cgi", cgiWiFiSetMode, NULL},
+
+	{"/websocket/ws.cgi", cgiWebsocket, myWebsocketConnect},
+	{"/websocket/echo.cgi", cgiWebsocket, myEchoWebsocketConnect},
+
+	{"/test", cgiRedirect, "/test/index.html"},
+	{"/test/", cgiRedirect, "/test/index.html"},
+	{"/test/test.cgi", cgiTestbed, NULL},
+
+	{"*", cgiEspFsHook, NULL}, //Catch-all cgi function for the filesystem
+	{NULL, NULL, NULL}
+};
+
+void wifiInit() {
+    struct ip_info ap_ip;
+    uint8_t sdk_wifi_get_opmode();
+    switch(sdk_wifi_get_opmode()) {
+        case STATIONAP_MODE:
+        case SOFTAP_MODE:
+            IP4_ADDR(&ap_ip.ip, 172, 16, 0, 1);
+            IP4_ADDR(&ap_ip.gw, 0, 0, 0, 0);
+            IP4_ADDR(&ap_ip.netmask, 255, 255, 0, 0);
+            sdk_wifi_set_ip_info(1, &ap_ip);
+
+            struct sdk_softap_config ap_config = {
+                .ssid = AP_SSID,
+                .ssid_hidden = 0,
+                .channel = 3,
+                .ssid_len = strlen(AP_SSID),
+                .authmode = AUTH_WPA_WPA2_PSK,
+                .password = AP_PSK,
+                .max_connection = 3,
+                .beacon_interval = 100,
+            };
+            sdk_wifi_softap_set_config(&ap_config);
+
+            ip_addr_t first_client_ip;
+            IP4_ADDR(&first_client_ip, 172, 16, 0, 2);
+            dhcpserver_start(&first_client_ip, 4);
+            break;
```

Comment:
These are needed:
```
dhcpserver_set_dns(&ap_ip.ip);
dhcpserver_set_router(&ap_ip.ip);
```

Reply:
Agree, this is a good addition, but it makes sense for some specific needs which are not covered in the example.
The example does not utilize any routing or NS resolution, that's why it does not set it.
Unless there is some kind of client which goes crazy when DNS and GW are not set, but I never seen it.

After all, this is just a minimal viable product, more or less required to get started.
Here is a good example of what I mean: https://github.com/SuperHouse/esp-open-rtos/blob/master/examples/access_point/access_point.c

Reply:
@nochkin fwiw I seem to recall Android not liking the link without a dns and/or router here - can't recall all the details.

Reply:
@ourairquality, that's interesting. I suspect it was old version or some other issue.
I remember I tried on Android 7 and 8 with no problems. I did not try anything older than that, but I think I could still locate some older device to test.

Reply:
I recently also tested this very same captive DNS code (https://github.com/SuperHouse/esp-open-rtos/pull/422) and it seemed that indeed setting the DNS and/or router field is required for Android 6.

Reply:
@flannelhead, thank you for providing more information on that. I updated my example to reflect that.